emcc.py

```diff
@@ -1600,9 +1600,16 @@ def compile_source_file(i, input_file):
           for tempf, inputf in zip(temp_files, input_files):
             safe_move(tempf[1], unsuffixed_basename(inputf[1]) + final_ending)
         else:
+          # We have a specified target (-o <target>), which is not JavaScript or HTML
           if len(input_files) == 1:
             input_file = input_files[0][1]
             temp_file = temp_files[0][1]
+            if not shared.Building.is_ar(input_file):
+              input_file = temp_file = None
+          else:
+            input_file = temp_file = None
+          if input_file is not None:
+            # only one input which is a bitcode and not an archive; simply copy it.
```

Comment:
@sbc100 Somehow `Building.is_ar` implies `Building.is_bitcode`.

That aside, I agree it indeed is convoluted and it should be fixed anyway since I now see it's more *correct* to test on `temp_file`, which is guaranteed to be either a bitcode or an archive.

Reply:
Hmm, that does look wrong. I think it may be an old assumption that an archive must contain bitcode files.

Reply:
@kripken I think at this point everything has been built with that assumption. When I tried to fix it by addredsing that things really broke.